src/ILCompiler.Compiler/src/Compiler/ExportedMethodsRootProvider.cs

```diff
@@ -12,12 +15,36 @@ namespace ILCompiler
     public class ExportedMethodsRootProvider : ICompilationRootProvider
     {
         private EcmaModule _module;
+        private List<EcmaMethod> _methods;
 
         public ExportedMethodsRootProvider(EcmaModule module)
         {
             _module = module;
         }
 
+        public IEnumerable<EcmaMethod> ExportedMethods
+        {
+            get
+            {
+                if (_methods != null)
+                    return _methods;
+
+                _methods = new List<EcmaMethod>();
```

Comment:
Nit: It is a better pattern to create the list in a local, and only assign it to the field once it is done. (Slightly better performance and reliability.)